pandas/core/frame.py

```diff
@@ -4847,7 +4847,7 @@ def _maybe_casted_values(index, labels=None):
                     if issubclass(values_type, DatetimeLikeArray):
                         values = values._data  # TODO: can we de-kludge yet?
 
-                    if mask.any():
+                    if mask.any() and isinstance(values, np.ndarray):
```

Comment:
slightly OT: the entire _maybe_casted_values should be moved to pandas/core/dtype/cast.py (separate issue)

Reply:
opened #36985